### PR TITLE
chore(CI): Ignore compliance container in pod restarts check

### PR DESCRIPTION
**Pull request by mtodor**
### Description

This PR is disabling the Pod Restarts check for the collector compliance container. After https://issues.redhat.com/browse/ROX-26311 - is resolved, we can revisit the ignore pattern.

We had such a pattern in the past, but after some compliance code refactoring, it became obsolete because the error message was changed (from `error ..` to `Error ..`).

I have "reused" that pattern slot and modified matching logging to also match `gke` tests.
I also fixed the `bats` test (btw. they are broken on Master, at least for me).

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [x] modified existing tests

#### How I validated my change

Executed `bats` test locally.

### CODE REVIEW

File: scripts/ci/logcheck/check-restart-logs.sh [hunk 1]
```diff
@@ -22,6 +22,7 @@ for logfile in "$@"; do
     fi
 done
 
+# TODO: revisit (update/delete) "compliance-previous" pattern after "ROX-26311" is resolved.
```
Comment (stehessel) [inline scripts/ci/logcheck/check-restart-logs.sh:25]:
```suggestion
# TODO(ROX-26311): revisit (update/delete) "compliance-previous" pattern after "ROX-26311" is resolved.
```
Let's mention in the ticket to revisit this comment once the ticket is resolved.

**Review by stehessel — APPROVED** [review]

**Review by vikin91 — APPROVED** [review]
The change looks good. I have not tested it, but I see that the fixture was used to cover this.

**Review by porridge — CHANGES_REQUESTED** [review]
LGTM overall, but I don't understand your comment about fixing bats tests. In fact they seem [broken *after* your change](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12815/pull-ci-stackrox-stackrox-master-shell-unit-tests/1839292412418068480#1:build-log.txt%3A140).

File: scripts/ci/logcheck/check-restart-logs.sh [hunk 2]
```diff
@@ -22,6 +22,7 @@ for logfile in "$@"; do
     fi
 done
 
+# TODO(ROX-26311): revisit (update/delete) "compliance-previous" pattern after "ROX-26311" is resolved.
```
[the remark this answers is not in the document]
Comment (mtodor) [reply scripts/ci/logcheck/check-restart-logs.sh:25]: @porridge I could remove this comment, and we can rely on `"comment": "ROX-26311: compliance is not able to establish connection with sensor",` - in JSON. Would that be ok?

The problem is that JSON format does not support comments. (at least to my knowledge)

[the remark this answers is not in the document]
Comment (porridge) [reply scripts/ci/logcheck/check-restart-logs.sh:25]: Would it be possible to change that to `"comment": "TODO(ROX-26311): compliance is not able to establish connection with sensor",` ?

**Comment by vikin91** [comment]
There is a fix candidate - #12859 - for the issue that lead to excluding the checks of compliance restarting in this PR.

**Comment by vikin91** [comment]
Please, let's hold merging this. I have just merged the https://github.com/stackrox/stackrox/pull/12859. The fix is non-trivial to validate outside of CI, so I would appreciate if we could keep the alerts about compliance crashing in CI.

**Comment by mtodor** [comment]
Moved to `draft` in favor of #12815 - will close it after the merge of another PR.

**Pull request closed without merging**